code/datums/datumvars.dm

```diff
@@ -850,8 +850,22 @@
 					A.create_reagents(amount)
 
 			if(A.reagents)
+				var/chosen_id
 				var/list/reagent_options = sortList(chemical_reagents_list)
-				var/chosen_id = input(usr, "Choose a reagent to add.", "Choose a reagent.") in reagent_options|null
+				switch(alert(usr, "Choose a method.", "Add Reagents", "Enter ID", "Choose ID"))
+					if("Enter ID")
+						var/valid_id
+						while(!valid_id)
+							chosen_id = stripped_input(usr, "Enter the ID of the reagent you want to add.")
+							if(!chosen_id) //Get me out of here!
+								break
+							for(var/ID in reagent_options)
+								if(ID == chosen_id)
+									valid_id = 1
+							if(!valid_id)
+								usr << "<span class='warning'>A reagent with that ID doesn't exist!</span>"
+					if("Choose ID")
+						chosen_id = input(usr, "Choose a reagent to add.", "Choose a reagent.") in reagent_options|null

```

Comment:
Mind doing `input() as null|anything in reagent_options` so it has a cancel button?